src/develop/imageop.h

```diff
@@ -731,8 +731,16 @@ gboolean dt_iop_show_hide_header_buttons(GtkWidget *header, GdkEventCrossing *ev
 /** Set the trouble message for the module.  If non-empty, also flag the module as being in trouble; if empty
  ** or NULL, clear the trouble flag.  If 'toast_message' is non-NULL/non-empty, pop up a toast with that
  ** message when the module does not have a warning-label widget (use %s for the module's name).  **/
-void dt_iop_set_module_trouble_message(dt_iop_module_t *module, char *const trouble_msg,
-                                       const char *const trouble_tooltip, const char *stderr_message);
+void dt_iop_set_module_trouble_message(dt_iop_module_t *module,
+                                       const char *const trouble_msg,
+                                       const char *const trouble_tooltip,
+                                       const char *stderr_message);
+
+void dt_iop_display_module_trouble_message_callback(gpointer instance,
+                                                    dt_iop_module_t *module,
```

Comment:
Does any code other than `dt_iop_set_module_trouble_message` even need to know that this callback exists?  That seems like the kind of implementation detail which is best hidden.

Reply:
So I could try to move that into darkroom implementation.

Reply:
I was just thinking that the callback function could be static inside imageop.c, and need not be declared in the header.

Reply:
Not possible as this is needed by darkroom implementation (in enter & cleanup).